azurerm/internal/services/apimanagement/api_management_certificate_resource.go

```diff
@@ -43,16 +45,32 @@ func resourceApiManagementCertificate() *schema.Resource {
 			"api_management_name": schemaz.SchemaApiManagementName(),
 
 			"data": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				Sensitive:     true,
+				ValidateFunc:  validation.StringIsBase64,
+				ConflictsWith: []string{"key_vault_secret_id", "key_vault_identity_client_id"},
+			},
+
+			"password": {
 				Type:         schema.TypeString,
-				Required:     true,
+				Optional:     true,
 				Sensitive:    true,
-				ValidateFunc: validation.StringIsBase64,
+				RequiredWith: []string{"data"},
 			},
 
-			"password": {
-				Type:      schema.TypeString,
-				Optional:  true,
-				Sensitive: true,
+			"key_vault_secret_id": {
+				Type:          schema.TypeString,
+				Optional:      true,
+				ValidateFunc:  keyVaultValidate.NestedItemIdWithOptionalVersion,
+				ConflictsWith: []string{"data", "password"},
+			},
+
+			"key_vault_identity_client_id": {
```

Comment:
doses it make sense to put these into a `key_vault` block?

Reply:
It makes sense, especially since that is how the request to API is done, but in my opinion for small resources like this it reduces the complexity if all properties are on top-level and we can handle conflicts between properties.
I don't see a lot of changes coming to this resource in the future after this is implemented.

Let me know if you want it changed.